### PR TITLE
custom 127 encoding

### PR DESCRIPTION
**Pull request by jlaustill**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=GL1sG8j)

### CODE REVIEW

File: src/unpack/unpack_0_127.ts [hunk 1]
```diff
@@ -30,12 +30,11 @@ export default function (incoming: Buffer, isoJSON: Types.KeyValueStringT, confi
...
     let thisBuff = incoming.slice(bitmapEnd, incoming.byteLength);
-    for (let i = 1; i < bitmap.length; i++) {
-      if (bitmap[i] === 1) {
+    for (let field = 2; field <= bitmap.length; field++) {
+      if (bitmap[field - 1] === 1) {
         // format defined
-        const field = i + 1;
         const this_format = this.formats[field] || formats[field];
-        if (!config.skip127 && field === 127) {
+        if (!config.custom127Encoding && field === 127) {
```
Comment (segevmalool) [inline src/unpack/unpack_0_127.ts:37]: What's the encoding?

Comment (segevmalool) [reply src/unpack/unpack_0_127.ts:37]: Oh got it.

**Review by segevmalool — APPROVED** [review]

File: src/unpack/unpack_0_127.ts (hunk 1 above)
Comment (erin-houston) [inline src/unpack/unpack_0_127.ts:37]: 127 should be a const With a meaningful name.

Comment (segevmalool) [reply src/unpack/unpack_0_127.ts:37]: Would the meaning would be derived from the iso8583 spec?

**Comment by segevmalool** [comment]
Do we need to commit the rebuild?

File: src/unpack/unpack_0_127.ts [hunk 2]
```diff
@@ -64,17 +63,20 @@ export default function (incoming: Buffer, isoJSON: Types.KeyValueStringT, confi
             if (thisLen === 0) {
               return T.toErrorObject(['field ', field, ' format not implemented']);
             } else {
-              const len = thisBuff.slice(0, thisLen).toString();
-              thisBuff = thisBuff.slice(thisLen, thisBuff.byteLength);
-              isoJSON[field] = thisBuff.slice(0, Number(len)).toString();
-              thisBuff = thisBuff.slice(Number(len), thisBuff.byteLength);
+              if (config.custom127Encoding && field === 127) {
+                isoJSON[field] = thisBuff.slice(0).toString(config.custom127Encoding);
```
Comment (segevmalool) [inline src/unpack/unpack_0_127.ts:67]: No updates needed to `thisBuff`?

Comment (jlaustill) [reply src/unpack/unpack_0_127.ts:67]: Nope, just taking the rest of the buffer as 127 encoded as hex

File: src/unpack/unpack_0_127.ts (hunk 1 above)
Comment (jlaustill) [reply src/unpack/unpack_0_127.ts:37]: 127 is already meaningful with the spec 🤷

Comment (erin-houston) [reply src/unpack/unpack_0_127.ts:37]: Not all people that work in here will know the spec.  It's about giving the next engineers a leg up.

File: src/unpack/unpack_0_127.ts [hunk 3]
```diff
@@ -64,17 +63,20 @@ export default function (incoming: Buffer, isoJSON: Types.KeyValueStringT, confi
...
+              } else {
+                const len = thisBuff.slice(0, thisLen).toString();
+                thisBuff = thisBuff.slice(thisLen, thisBuff.byteLength);
+                isoJSON[field] = thisBuff.slice(0, Number(len)).toString(config.bitmapEncoding || 'hex');
+                thisBuff = thisBuff.slice(Number(len), thisBuff.byteLength);
```
Comment (erin-houston) [inline src/unpack/unpack_0_127.ts:72]: should be a sub function returnigin an arra [value, tailBuff ]
would probably make it easier to read and you could write some meaningful tests for that.

**Pull request merged**